### DISCUSSION TITLE
How do we console.log to the Browser developer tools for NET MAUI Blazor?

### DISCUSSION THREAD

**Discussion by Assassinbeast** [dotnet/maui#6091] [Q&A] [answered]
I read this doc https://docs.microsoft.com/en-us/aspnet/core/blazor/hybrid/developer-tools?view=aspnetcore-6.0&pivots=windows that we can open the Developer tools with Ctrl+Shift+I and see the Console logs.

But how do we write to the logs?

I have tried to add the `builder.Logging.AddConsole();` and `builder.Logging.AddDebug();`, to MauiProgram, but it didn't work.
Below code is what i tried:

```
public static class MauiProgram
{
	public static MauiApp CreateMauiApp()
	{
		var builder = MauiApp.CreateBuilder();
		builder
			.UseMauiApp<App>()
			.ConfigureFonts(fonts =>
			{
				fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
			});

		builder.Services.AddMauiBlazorWebView();
		builder.Services.AddBlazorWebViewDeveloperTools();
		builder.Logging.AddConsole();
		builder.Logging.AddDebug();

		builder.Services.AddSingleton<WeatherForecastService>();

		return builder.Build();
	}
}
```

And then i added the _logger.LogInformation() to the Counter.razor

```
@using Microsoft.Extensions.Logging
@page "/counter"
@inject ILogger<Counter> _logger

<h1>Counter</h1>

<p role="status">Current count: @currentCount</p>

<button class="btn btn-primary" @onclick="IncrementCount">Click me</button>

@code{
    	private int currentCount = 0;

		private void IncrementCount()
		{
			currentCount++;
			_logger.LogInformation(currentCount.ToString());
		}
}
```

Then i try to click on the "Click me" button, but there are no logs writing to the console

![mauilogging](https://user-images.githubusercontent.com/6762656/163294183-e2255bff-8768-42eb-967b-82132d6c0e17.jpg)

**Comment by Assassinbeast** [comment]
cc: @guardrex https://github.com/dotnet/AspNetCore.Docs/issues/25625

**Comment by TanayParikh** [comment, marked as answer]
@Assassinbeast, thanks for reaching out. Can you just try the trusty `Console.WriteLine`, I believe that should do the trick 😄

**Reply by Assassinbeast** [reply]
Hi @TanayParikh :)  , i just tried `Console.WriteLine` it didn't work though.

**Reply by TanayParikh** [reply]
Ah I see, I'm seeing the same. I was able to get logging working via the following, please give it a try!

```razor
@inject IJSRuntime JSRuntime
```

and then

```razor
_ = JSRuntime.InvokeVoidAsync("console.log", "some js runtime log");
```

**Comment by guardrex** [comment]
:smile: on that one, @TanayParikh ... I guessed something like that would be the answer. BTW ... We don't document that approach for ASP.NET Core generally in the main doc set or specifically for Blazor Server, so we might want to call it out in the Blazor *Fundamentals* logging topic and in the .NET MAUI DevTools topic ... or not ... you can let me know on https://github.com/dotnet/AspNetCore.Docs/issues/25625 how you want to proceed.

**Reply by TanayParikh** [reply]
Let's put that on ice for now @guardrex, I'd like to confirm a few additional details. Let me get back to you on this.

**Reply by guardrex** [reply]
Very well. I'll close the docs issue because I'm monitoring the discussion here. I know where to point devs if more docs repo issues come in on this.

**Reply by TanayParikh** [reply]
@guardrex a longer term fix for this issue will be tracked here: https://github.com/dotnet/maui/issues/6110
